pkg/errors/kubeclient.go

```diff
@@ -0,0 +1,72 @@
+package errors
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type apiVerbs struct {
+	Create   string
+	Get      string
+	Update   string
+	Delete   string
+	Discover string
+	List     string
+}
+
+var API = &apiVerbs{
+	Create:   "CREATE",
+	Get:      "GET",
+	Update:   "UPDATE",
+	Delete:   "DELETE",
+	Discover: "DISCOVER",
+	List:     "LIST",
+}
+
+type FailedAPICall struct {
+	verb     string
+	resource string
+	err      error
+}
+
+func (e FailedAPICall) Error() string {
+	return fmt.Sprintf(
+		"failed API call: %s %s: %q", e.verb, e.resource, e.err,
+	)
+}
+
+// ShouldRetry determines if the error generated by an API call will be
+// retried. It follows API conventions stipulated by
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#http-status-codes
+func (e FailedAPICall) ShouldRetry() bool {
+	switch kerrors.ReasonForError(e) {
```

Comment:
Go's switch cases do not fall through (e.g. like in C). What you want here is comma-separated list of values. And a test.